rfcs/0000-rfc-session.md

```diff
@@ -0,0 +1,205 @@
+# 0000: Session
+<!--^ The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC, taking care not to conflict with other RFCs.-->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage -->
+- Date: 6/22/2020 <!-- Update to reflect date of most recent stage advancement -->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+## Fields
+This RFC calls for the addition of session fields to describe events related to various types of "sessions" reported by appliances, security devices, systems, management portals, applications, etc.
+
+| Field | Description |
+| ----- | ----------- |
+|session.kind:            | local, remote, network
+|session.authorization:   | user, admin, service
+|session.type:            | system, virtual, application, wired, wireless, vpn
+|session.name             | locally relevant name if available (e.g. HQ Client VPN, Win19-VDI, FIN-EXCEL-vApp)
+|session.id               | session id provided by server or custom fingerprint
+
+
+
+## Fields (yaml)
+```yaml
+---
+- name: session
+  title: Session
+  group: 2
+  short: User, admin, application, network, or service sessions
+  description: |-
+    These fields contain information about various types of user sessions typically reported & logged in an enterprise.
```

Comment:
:grey_question: This reads a little bit like a tautology ( "A session is something that is typically a session"  ).  Adding a more formal definition like "A session typically encompasses the data that a system uses to manage instances of authentication against user identities, roles, etc."

Reply:
Excellent suggestion - will work it in.

Reply:
updated

```
These fields are used to track an entity's interaction with various assets, services, and applications in an enterprise.  Sessions will typically include a start event, often a login / authorization event performed locally or via network based mechanisms, and an end event indicating a logoff or session termination.  

When available, event start/end or duration fields should be populated, as well as iam, user, network, host, observer, process, source, destination, client, and server fields as appropriate to describe the specifics of the interaction.
```